src/rgw/rgw_rados.cc

```diff
@@ -4292,6 +4292,32 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
   if (cmp != src_attrs.end())
     attrs[RGW_ATTR_COMPRESSION] = cmp->second;
 
+  attrs.erase(RGW_ATTR_OBJECT_RETENTION);
```

Comment:
it seems like a problem that we are doing this in RGWRados, due to Zipper refactoring;  @dang does that sound right to you?

Reply:
I agree, this belongs in the Op layer, in RGWCopyObj::execute()